messages/run.csx

```diff
@@ -0,0 +1,80 @@
+
+// (Default) For Azure Table Storage, set the following environment variables in your bot app:
+// -UseTableStorageForConversationState set to 'true'
+// -AzureWebJobsStorage set to your table connection string
+
+// For CosmosDb, set the following environment variables in your bot app:
+// -UseCosmosDbForConversationState set to 'true'
+// -CosmosDbEndpoint set to your cosmos db endpoint
+// -CosmosDbKey set to your cosmos db key
+
+public static async Task<object> Run(HttpRequestMessage req, TraceWriter log)
+{
+    log.Info($"Webhook was triggered!");
+
+    // Initialize the azure bot
+    using (BotService.Initialize())
+    {
+        // Deserialize the incoming activity
+        string jsonContent = await req.Content.ReadAsStringAsync();
+        var activity = JsonConvert.DeserializeObject<Activity>(jsonContent);
+        
+        // authenticate incoming request and add activity.ServiceUrl to MicrosoftAppCredentials.TrustedHostNames
+        // if request is authenticated
+        if (!await BotService.Authenticator.TryAuthenticateAsync(req, new [] {activity}, CancellationToken.None))
+        {
+            return BotAuthenticator.GenerateUnauthorizedResponse(req);
+        }
+        
+        if (activity != null)
+        {
+            // one of these will have an interface and process it
+            switch (activity.GetActivityType())
+            {
+                case ActivityTypes.Message:
+                    await Conversation.SendAsync(activity, () => new EchoDialog());
+                    break;
+                case ActivityTypes.ConversationUpdate:
+                    // var client = new ConnectorClient(new Uri(activity.ServiceUrl));
+                    // IConversationUpdateActivity update = activity;
+                    // if (update.MembersAdded.Any())
+                    // {
+                    //     var reply = activity.CreateReply();
+                    //     var newMembers = update.MembersAdded?.Where(t => t.Id != activity.Recipient.Id);
+                    //     foreach (var newMember in newMembers)
+                    //     {
+                    //         reply.Text = "Welcome";
+                    //         if (!string.IsNullOrEmpty(newMember.Name))
+                    //         {
+                    //             reply.Text += $" {newMember.Name}";
+                    //         }
+                    //         reply.Text += "!";
+                    //         await client.Conversations.ReplyToActivityAsync(reply);
+                    //     }
+                    // }
+                    // break;
+                case ActivityTypes.ContactRelationUpdate:
+                case ActivityTypes.Typing:
+                case ActivityTypes.DeleteUserData:
+                case ActivityTypes.Ping:
+                default:
+                    log.Error($"Unknown activity type ignored: {activity.GetActivityType()}");

```

Comment:
We expect to get these activity types all the time in a Teams bot, no need to log an error

Reply:
removed